docs/misc/keymappings.md

```diff
@@ -0,0 +1,150 @@
+)
+```
+
+!!! note
+    If the key mapping should not be mapped to a default, the input should be set to `InputConstants#UNKNOWN`. The vanilla constructor will require you to extract the input code via `#getValue` while the Forge constructor can be supplied the raw input field.
+
+### `IKeyConflictContext`
+
+Not all mappings are used in every context. Some mappings are only used in a GUI, while others are only used purely in game. To avoid mappings of the same key used in different contexts conflicting with each other, an `IKeyConflictContext` can be assigned.
+
+Each conflict context contains two methods: `#isActive`, which defines if the mapping can be used in the current game state, and `#conflicts`, which defines whether the mapping conflicts with a key in the same or different conflict context.
+
+Currently, Forge defines three basic contexts through `KeyConflictContext`: `UNIVERSAL`, which is the default meaning the key can be used in every context, `GUI`, which means the mapping can only be used when a `Screen` is open, and `IN_GAME`, which means the mapping can only be used if a `Screen` is not open. New conflict contexts can be created by implementing `IKeyConflictContext`.
+
+```java
+new KeyMapping(
+  "key.examplemod.example2",
+  KeyConflictContext.GUI, // Mapping can only be used when a screen is open
+  InputConstants.Type.MOUSE, // Default mapping is on the mouse
+  GLFW.GLFW_MOUSE_BUTTON_LEFT, // Default mouse input is the left mouse button
+  "key.categories.examplemod.custom" // Mapping will be in the new custom category
+)
+```
+
+### `KeyModifier`
+
+Modders may not want mappings to have the same behavior if a modifier key is held at the same (e.g. `G` vs `CTRL + G`). To remedy this, Forge adds an additional parameter to the constructor to take in a `KeyModifier` which can apply control (`KeyModifier#CONTROL`), shift (`KeyModifier#SHIFT`), or alt (`KeyModifier#ALT`) to any input. `KeyModifier#NONE` is the default and will apply no modifier.
+
+A modifier can be added in the [controls option menu][controls] by holding down the modifier key and the associated input.
+
+```java
+new KeyMapping(
+  "key.examplemod.example3",
+  KeyConflictContext.UNIVERSAL,
+  KeyModifier.SHIFT, // Default mapping requires shift to be held down
+  InputConstants.Type.KEYSYM, // Default mapping is on the keyboard
+  GLFW.GLFW_KEY_G, // Default key is G
+  "key.categories.misc"
+)
+```
+
+## Checking a `KeyMapping`
+
+A `KeyMapping` can be checked to see whether it has been clicked. Depending on when, the mapping can be used in a conditional to apply the associated logic.
+
+### Within the Game
+
+Within the game, a mapping should be checked by listening to `ClientTickEvent` on the [**forge event bus**][forgebus] and checking `KeyMapping#consumeClick` within a while loop. `#consumeClick` will return `true` only the number of times the input was performed and not already previously handled, so it won't infinitely stall the game.
```

Comment:
"forge" should be capitalized. Same should be done in further references to the Forge event bus.
```suggestion
Within the game, a mapping should be checked by listening to `ClientTickEvent` on the [**Forge event bus**][forgebus] and checking `KeyMapping#consumeClick` within a while loop. `#consumeClick` will return `true` only the number of times the input was performed and not already previously handled, so it won't infinitely stall the game.
```